docs/style/style-guide.md

```diff
@@ -28,16 +22,106 @@ Meeting this goal will require your careful consideration about when, where, and
-    - If the method can return ```null``` it should be noted and under what condition this may happen
-  - Method overrides or interface implementations should have a javadoc that is simply ```{@inheritDoc}``` unless the overridden method's functionality diverges enough to need additional documentation
-    - This includes ```toString```, ```hashCode```, etc.
-- All classes and interfaces of all scopes must have a javadoc 
-- All class-level fields of all scopes must have a javadoc
-    - Information about the field should be captured in the field javadoc, which can be referenced by the getter/setter to reduce duplication
-    - Reference other classes when it makes sense to reduce duplication (for instance, a class level field that represents a CcwCodebookVariable can just link to the CcwCodebookVariable in the documentation)
+**Classes**
+
+Classes must have a description; ideally this should be a high-level description of the class responsibility and/or purpose that is at least one full sentence ending in a period.
+
+```
+/** Orchestrates and manages the execution of {@link PipelineJob}s. */
+public final class PipelineManager implements AutoCloseable {
+...
+```
+
+**Class Fields** 
+
+Fields must have a full-sentence description which should describe what the purpose of the field is, and contain useful contextual information if applicable.
+- Information about the field should be captured in the field javadoc, which can be referenced by the getter/setter to reduce duplication
+
+```
+/** Holds the records of completed jobs. */
+private final PipelineJobRecordStore jobRecordsStore;
+```
+
+**Methods**
+
+Methods must have at least a one sentence description of what the method does, and must end in a period. It should avoid being too descriptive of the implementation
+details, but still describe what the method does.
+ - All method parameters should be documented with the ```@param``` tag, and a description of what the parameter is
+ - Method parameter descriptions don't need to be full sentences and should not end in punctuation, they can be short descriptions
+ - If ```null``` is explicitly disallowed for a parameter, it should be documented
+ - Method overrides or interface implementations should have a javadoc that is simply ```{@inheritDoc}``` unless the overridden method's functionality diverges enough to need additional documentation
+   - This includes ```toString```, ```hashCode```, etc.
+ - Method returns should be documented with the ```@return``` tag and a description of what is returned (and how if multiple return conditions under various conditions)
+   - If the method can return ```null``` it should be noted and under what condition this may happen
+   - If the method returns a boolean, you can just list the ```true``` case (the false case should an obvious inverse, but if not document it)
```

Comment:
```suggestion
 - All method parameters should be documented with the `@param` tag, and a description of what the parameter is
 - Method parameter descriptions don't need to be full sentences and should not end in punctuation, they can be short descriptions
 - If `null` is explicitly disallowed for a parameter, it should be documented
 - Method overrides or interface implementations should have a javadoc that is simply `{@inheritDoc}` unless the overridden method's functionality diverges enough to need additional documentation
   - This includes `toString`, `hashCode`, etc.
 - Method returns should be documented with the `@return` tag and a description of what is returned (and how if multiple return conditions under various conditions)
   - If the method can return `null` it should be noted and under what condition this may happen
   - If the method returns a boolean, you can just list the `true` case (the false case should an obvious inverse, but if not document it)
```